src/Servers/Kestrel/Core/src/Internal/Http/ChunkWriter.cs

```diff
@@ -6,12 +6,13 @@
 using System.IO.Pipelines;
 using System.Text;
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
 {
     internal static class ChunkWriter
     {
-        private static readonly byte[] _hex = Encoding.ASCII.GetBytes("0123456789abcdef");
+        private static ReadOnlySpan<byte> Hex => new byte[16] { (byte)'0', (byte)'1', (byte)'2', (byte)'3', (byte)'4', (byte)'5', (byte)'6', (byte)'7', (byte)'8', (byte)'9', (byte)'a', (byte)'b', (byte)'c', (byte)'d', (byte)'e', (byte)'f' };
```

Comment:
This reads like it would create a new byte[] every time it was called. What obscure syntax rule would prevent that?

Reply:
Shall I add your link as comment?

Reply:
Please do! To all the places we use it. Otherwise someone is going to come along and try to "fix it".

Reply:
Also, if you could remove the size from the array declaration that would make it a little easier to maintain.

Reply:
I'll add these comments presumably tomorrow.

> remove the size from the array declaration

Hm, I put them intentionally as "safety". Otherwise it's easy to have an element more or less. So the C# compiler would complain if the element count doesn't match.

Reply:
If the compiler will complain, then maybe it's OK. I'm not sure it's "safe" though. The value is the value, the size of it is just repetition.

Reply:
In my experience the (redundant) length gives additional safety. E.g.
```c
ReadOnlySpan<byte> fourItemsData = new byte[] { 0x1, 0x2, 0x3 };
ReadOnlySpan<byte> saferFourItemsData = new byte[4] { 0x1, 0x2, 0x3 };   // compiler error
```
both are easy to type, but the longer they get, the easier to forget an item or put an item too much in it.

In other words: I got trapped a couple of times by this, so I started to put the length to the initializer.

Anyway, it's up to your decision what I shall do with them :wink:

Reply:
@jaredpar Can you fix this please? We need better looking constants.

Reply:
Fair enough :)